src/gpu/optixPrograms.cu

```diff
@@ -139,10 +136,20 @@ extern "C" __global__ void __closesthit__()
 		optixGetPayload_2()
 	});
 
+	float distance = sqrt(pow((hitWorld)[0] - (origin)[0], 2) +
```

Comment:
We can think of and discuss the little neat trick.
We can use Inversesqrt here, which is one operation cheaper then sqrt.
This will cost us passing inverse distance to the shader, however I think we should think about details like that on Our way to 1B rays/second.

Reply:
Good point for the optimization task. I will leave a TODO comment here.